### DISCUSSION TITLE
I can't use yup for schema validation because it somehow conflicts with defaultValues!!!

### DISCUSSION THREAD

**Discussion by amirhhashemi** [react-hook-form/react-hook-form#6702] [Q&A] [answered]
I recently migrated from formik to react-hook-form But I couldn't create a simple form I get wired type errors when using yup for schema validation.

this works:
```
  const { register, handleSubmit } = useForm({
    defaultValues: {
      email: "",
      password: "",
      passwordConfirmation: "",
      policy: false,
    },
  });
```
but as soon as I add this line:
`resolver: yupResolver(validationSchema)`
I get this error on defaultValues:
```
[tsserver 2322] [E] Type '{ email: string; password: string; passwordConfirmation: string; policy: false; }' is not assignable to type '{ [x: string]: undefined; email?: string | undefined; password?: string | undefined; passwordConfirmation?: string | undefined; policy?: boolean | undefined; }'.
  Property 'email' is incompatible with index signature.
    Type 'string' is not assignable to type 'undefined'.
```
also I get some errors in my inputs. for example this:
`<input {...register("email")} />`
gives me this error:
`[tsserver 2345] [E] Argument of type 'string' is not assignable to parameter of type 'never'.`
this is the validationSchema:
```
const validationSchema = Yup.object().shape({
  email: Yup.string().email("Invalid email").required("Required"),
  password: Yup.string()
    .min(4, "Password should be at least 4 characters")
    .max(20, "Password can't be more than 20 characters")
    .required("Required"),
  passwordConfirmation: Yup.string()
    .required("Required")
    .oneOf([Yup.ref("password")], "Passwords must match"),
  policy: Yup.boolean().oneOf([true], "Required"),
});
```

I don't know what is wrong with my validation schema. can you help me?

**Comment by bluebill1049** [comment, marked as answer]
can you double-check the doc on the usage: https://react-hook-form.com/api/useform?

**Reply by amirhhashemi** [reply]
I did. 
I saw some people directly use the `validationSchema` field in the useForm config. apparently, it's not the case.
anyway. I changed my question.

**Reply by bluebill1049** [reply]
provide a codesandbox.

**Reply by amirhhashemi** [reply]
https://codesandbox.io/s/affectionate-kalam-ez95t?file=/src/App.tsx

**Reply by bluebill1049** [reply]
https://codesandbox.io/s/gifted-elbakyan-mo3g9

**Reply by amirhhashemi** [reply]
Thank you, sir. so I just need to make the schema required by adding `.required()` at the end of schema decoration.

**Reply by bluebill1049** [reply]
for more information, please refer to this comment: https://github.com/react-hook-form/resolvers/issues/245#issuecomment-924647399 It's a better approach.

**Reply by amirhhashemi** [reply]
Thanks. I will check it out.
